internal/configconverter/update_batchproc_on_token_passthrough.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+func UpdateBatchProcOnTokenPassthrough(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return nil
+	}
+
+	out := map[string]any{}
+
+	// Check for sapm receiver with include_metadata set to true
+	if in.IsSet("receivers::sapm::include_metadata") {
+		if _, ok := in.Get("receivers::sapm::include_metadata").(bool); ok {
```

Comment:
`ok` represents whether the value casts to booled. So it's going to be true even for `include_metadata: true`. That wasn't your intention, right? Please fix and add a test case

Reply:
makes sense. I'll update it